packages/core/src/actors/promise.ts

```diff
@@ -88,8 +88,11 @@ export function fromPromise<TOutput, TInput = NonReducibleUnknown>(
      * The parent actor of the promise actor
      */
     self: PromiseActorRef<TOutput>;
+    signal: AbortSignal;
   }) => PromiseLike<TOutput>
 ): PromiseActorLogic<TOutput, TInput> {
+  const controller = new AbortController();
+  let done = false;
```

Comment:
This introduces shared state between all of the actors created based on this promise logic. Each actor should get its own controller/signal - it would also be great to add a test case for this.

You can use this is the starting point for this test case. The second region should be completed when we cancel the first one
```ts
const p = fromPromise(() => { /* ... */ })
createMachine({
  type: 'parallel',
  states: {
    p1: {
      initial: 'running',
      states: {
        running: {
          invoke: {
            src: p
          },
          on: {
            CANCEL_1: 'canceled'
          }
        },
        canceled: {}
      }
    },
    p2: {
      initial: 'running',
      states: {
        running: {
          invoke: {
            src: p,
            onDone: 'done'
          },
        },
        done: {}
      }
    }
  }
})
```

Reply:
[`c06b029` (#4832)](https://github.com/statelyai/xstate/pull/4832/commits/c06b029019d1aac6d9d66166a5a940c4f49661f8)